contracts/Blacksmith.sol

```diff
@@ -129,17 +131,32 @@ contract Blacksmith is Initializable, AccessControlUpgradeable {
         IERC20(tokenAddress).safeTransfer(msg.sender, amount);
     }
 
+    function setSafeRandoms(SafeRandoms _safeRandoms) public isAdmin {
+        safeRandoms = _safeRandoms;
+    }
+
+    function hasSeed() public view returns (bool) {
```

Comment:
May be forward thinking to have a seed ID param on this function for frontend compatibility. Assuming we'll sell more seed-bound things in blacksmith in the future, so the ABI doesnt have to change. The frontend would have to know about the seed id constant though.